recipe-client-addon/test/browser/browser_RecipeRunner.js

```diff
@@ -212,25 +212,37 @@ add_task(async function testStartup() {
   const addCleanupHandlerStub = sinon.stub(CleanupManager, "addCleanupHandler");
   const updateRunIntervalStub = sinon.stub(RecipeRunner, "updateRunInterval");
 
-  // in dev mode
-  await SpecialPowers.pushPrefEnv({set: [["extensions.shield-recipe-client.dev_mode", true]]});
-  RecipeRunner.init();
-  ok(runStub.called, "RecipeRunner.run is called immediately when in dev mode");
-  ok(addCleanupHandlerStub.called, "A cleanup function is registered when in dev mode");
-  ok(updateRunIntervalStub.called, "A timer is registered when in dev mode");
-
-  runStub.reset();
-  addCleanupHandlerStub.reset();
-  updateRunIntervalStub.reset();
-
-  // not in dev mode
-  await SpecialPowers.pushPrefEnv({set: [["extensions.shield-recipe-client.dev_mode", false]]});
-  RecipeRunner.init();
-  ok(!runStub.called, "RecipeRunner.run is not called immediately when not in dev mode");
-  ok(addCleanupHandlerStub.called, "A cleanup function is registered when not in dev mode");
-  ok(updateRunIntervalStub.called, "A timer is registered when not in dev mode");
-
-  runStub.restore();
-  addCleanupHandlerStub.restore();
-  updateRunIntervalStub.restore();
+  const UI_AVAILABLE_NOTIFICATION = "sessionstore-windows-restored";
+  const observer = {
+    observe: async function(subject, topic, data) {
+      Services.obs.removeObserver(this, UI_AVAILABLE_NOTIFICATION);
+
+      // in dev mode
+      await SpecialPowers.pushPrefEnv({set: [["extensions.shield-recipe-client.dev_mode", true]]});
+      RecipeRunner.init();
+      ok(runStub.called, "RecipeRunner.run is called immediately when in dev mode");
+      ok(addCleanupHandlerStub.called, "A cleanup function is registered when in dev mode");
+      ok(updateRunIntervalStub.called, "A timer is registered when in dev mode");
+
+      runStub.reset();
+      addCleanupHandlerStub.reset();
+      updateRunIntervalStub.reset();
+
+      // not in dev mode
+      await SpecialPowers.pushPrefEnv({set: [["extensions.shield-recipe-client.dev_mode", false]]});
+      RecipeRunner.init();
+      ok(!runStub.called, "RecipeRunner.run is not called immediately when not in dev mode");
+      ok(addCleanupHandlerStub.called, "A cleanup function is registered when not in dev mode");
+      ok(updateRunIntervalStub.called, "A timer is registered when not in dev mode");
+
+      runStub.restore();
+      addCleanupHandlerStub.restore();
+      updateRunIntervalStub.restore();
+    },
+    QueryInterface: XPCOMUtils.generateQI([
+      Components.interfaces.nsIObserver,
+      Components.interfaces.nsISupportsWeakReference,
+    ]),
+  };
+  Services.obs.addObserver(observer, UI_AVAILABLE_NOTIFICATION);
```

Comment:
I have to admit I don't understand this change. Surely there's no guarantee that the test runs after this observer notification has been sent? Where is it being sent instead?